src/parser/jobs/sch/modules/DoTs.js

```diff
@@ -28,12 +29,140 @@ export default class DoTs extends CoreDoTs {
 		...DoTs.dependencies,
 		'checklist',
 		'suggestions',
+		'downtime',
 	]
 
 	static statusesToTrack = [
 		STATUSES.BIOLYSIS.id,
 	]
 
+	_lastBioCast = undefined
+	_application = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const castFilter = {
+			by: 'player',
+			abilityId: ACTIONS.BIOLYSIS.id,
+		}
+		this.addEventHook('cast', castFilter, this._onTrackedDotCast)
+		const statusFilter = {
+			by: 'player',
+			abilityId: [STATUSES.BIOLYSIS.id],
+		}
+		this.addEventHook(['applydebuff', 'refreshdebuff'], statusFilter, this._onTrackedDotApply)
+	}
+
+	_pushApplication(targetKey, statusId, event) {
+		const target = this._application[targetKey] = this._application[targetKey] || {[STATUSES.BIOLYSIS.id]: []}
+		const source = this._lastBioCast
+		target[statusId].push({event, source})
+	}
+
+	_onTrackedDotCast(event) {
+		this._lastBioCast = event.ability.guid
+	}
+
+	_onTrackedDotApply(event) {
+		const statusId = event.ability.guid
+
+		// Make sure we're tracking for this target
+		const applicationKey = `${event.targetID}|${event.targetInstance}`
+		//save the application for later use in the output
+		this._pushApplication(applicationKey, statusId, event)
+	}
+
+	_formatDriftText(timestamp, previous=0) {
+		let drift = 0
+		const lateAllow = 1000
+		const earlyAllow = -2500
+		if (previous !== 0) {
+			const delta = timestamp - previous
+			drift = delta - (STATUSES.BIOLYSIS.duration * 1000)
+			if (drift > (STATUSES.BIOLYSIS.duration * 1000)) {
```

Comment:
As written, this isn't going to give any forgiveness in a case such as:
- Applied Biolysis 20 seconds before the boss went untargetable
- Boss goes untargetable for 20 seconds
- DOT is reapplied immediately after the Boss becomes available (so roughly 10 seconds after it "fell off" while the boss was untargetable).

The "use your cooldowns" module adjusts for downtime by moving the "time of next expected use" to the end of the downtime window - would that make sense here?  So in the above example, the drift calculation would update "previous" to be the end of the downtime window, maybe plus a GCD of forgiveness in case there's something else priority to cast after the downtime?

Reply:
sorry, but where is the use your cooldowns module? is it cooldowns or cooldowndowntime?

Reply:
The module name in core is CooldownDowntime, yes.  Which results in the "Use your cooldowns" checklist item for most classes.